cc/concurrent_selfplay.cc

```diff
@@ -533,18 +556,31 @@ bool SelfplayGame::MaybePlayMove() {
       model_str = absl::StrCat("model: ", models_used_.back(), "\n");
     }
 
-    auto search_pi = tree_->CalculateSearchPi();
-    game_->AddMove(tree_->to_play(), c, tree_->root()->position,
-                   std::move(model_str), tree_->root()->Q(), search_pi);
+    if (!fastplay_ && c != Coord::kResign) {
+      auto search_pi = tree_->CalculateSearchPi();
+      game_->AddTrainableMove(tree_->to_play(), c, tree_->root()->position,
+                              std::move(model_str), tree_->root()->Q(),
+                              search_pi);
+    } else {
+      game_->AddNonTrainableMove(tree_->to_play(), c, tree_->root()->position,
+                                 std::move(model_str), tree_->root()->Q());
+    }
 
     if (options_.target_pruning && !fastplay_) {
-      tree_->ReshapeFinalVisits();
+      tree_->ReshapeFinalVisits(restrict_pass_alive_moves);
```

Comment:
Don't you need to reshape it before you add it as a trainable position?

Reply:
oof, thanks for catching that